### DISCUSSION TITLE
Raising VauleError inside field_validator is not working properly with Form field

### DISCUSSION THREAD

**Discussion by EvgeniyMoiseevUnifonic** [fastapi/fastapi#10779] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Annotated

from fastapi import FastAPI, Depends, UploadFile, Form
from pydantic import BaseModel, field_validator

app = FastAPI()


class UploadLogoInScheme(BaseModel):
    image: UploadFile = Form()

    @field_validator('image')
    @classmethod
    def validate_image(cls, v: UploadFile) -> UploadFile:
        if not v.filename.lower().endswith(('.png', '.jpg', '.jpeg')):
            raise ValueError('Invalid file extension. It should be .png, .jpg or .jpeg')
        return v


@app.post("/logo")
async def upload_logo(payload: Annotated[UploadLogoInScheme, Depends()]) -> None:
    return
```

### Description

I want to validate some parameters of the uploaded image within form data. For this, I'm trying to use a standard scheme with raising `ValueError` inside the validating method decorated with `field_validator`.  But instead of returning the expected validation error with the 422 response status code it returns 500 as a result of raising ValueError inside Pydantic like proper RequestValidationError was never raised. I'm pretty sure this is not how it should be.

curl:
```
curl --location 'localhost:8000/logo' \
--form 'image=@"/Users/emoiseev/Downloads/invalid_extension.jpq"'
```

### Operating System

macOS

### Operating System Details

23.1.0 Darwin Kernel Version 23.1.0: Mon Oct  9 21:28:31 PDT 2023; root:xnu-10002.41.9~6/RELEASE_ARM64_T8112 arm64

### FastAPI Version

0.103.2

### Pydantic Version

2.4.2

### Python Version

3.11.5

**Comment by JavierSanchezCastro** [comment]
Hi!

Try this:

```python3
from typing import Annotated

from fastapi import FastAPI, Depends, UploadFile, Form, HTTPException, status
from pydantic import BaseModel, field_validator

app = FastAPI()


class UploadLogoInScheme(BaseModel):
    image: UploadFile = Form()

    @field_validator('image')
    @classmethod
    def validate_image(cls, v: UploadFile) -> UploadFile:
        if not v.filename.lower().endswith(('.png', '.jpg', '.jpeg')):
            raise HTTPException(status_code=status.HTTP_422_UNPROCESSABLE_ENTITY, detail='Invalid file extension. It should be .png, .jpg or .jpeg')
        return v


@app.post("/logo")
async def upload_logo(payload: Annotated[UploadLogoInScheme, Depends()]) -> None:
    return
```

https://github.com/tiangolo/fastapi/issues/3761
https://stackoverflow.com/questions/68914523/fastapi-pydantic-value-error-raises-internal-server-error
https://fastapi.tiangolo.com/tutorial/handling-errors/?h=validation#requestvalidationerror-vs-validationerror
https://github.com/tiangolo/fastapi/issues/1474#issuecomment-803021344
https://github.com/tiangolo/fastapi/issues/1474#issuecomment-1049987786

**Reply by EvgeniyMoiseevUnifonic** [reply]
@JavierSanchezCastro thanks for the links, I'll close this one as duplicate
